cpp/include/kvikio/posix_io.hpp

```diff
@@ -84,43 +104,44 @@ class StreamsByThread {
 /**
  * @brief Read or write host memory to or from disk using POSIX
  *
- * @tparam IsReadOperation Whether the operation is a read or a write
+ * @tparam ioType Whether the operation is a read or a write.
+ * @tparam ioDataCompletionLevel Whether all requested data are processed or not. If `FULL`, all of
+ * `count` bytes are read or written.
  * @param fd File descriptor
  * @param buf Buffer to write
  * @param count Number of bytes to write
  * @param offset File offset
- * @param partial If false, all of `count` bytes are read or written.
  * @return The number of bytes read or written (always gather than zero)
  */
-template <bool IsReadOperation>
-ssize_t posix_host_io(int fd, const void* buf, size_t count, off_t offset, bool partial)
+template <IOType ioType, IODataCompletionLevel ioDataCompletionLevel>
```

Comment:
Is there really a benefit to templating on `IODataCompletionLevel`? It adds a single break to the loop which should be predicted correctly every time by the branch predictor.

Reply:
> which should be predicted correctly every time by the branch predictor.

@wence-: ELI5 how we know that that would be predicted correctly?  (Asking out of ignorance.)

Would that not also apply to `IsReadOperation` in the prior/original version?

Reply:
My intention was to make the interface more consistent. Either `IsReadOperation` and  `partial` are both template parameters or both function parameters.